rust/src/serialization.rs

```diff
@@ -1726,147 +1715,153 @@ impl DeserializeEmbeddedGroup for MsigPubkey {
                 return Err(DeserializeFailure::FixedValueMismatch{ found: Key::Uint(index_0_value), expected: Key::Uint(1) }.into());
             }
             Ok(())
         })().map_err(|e| e.annotate("index_0"))?;
-        let multisig_scripts = (|| -> Result<_, DeserializeError> {
-            Ok(MultisigScripts::deserialize(raw)?)
-        })().map_err(|e| e.annotate("multisig_scripts"))?;
-        Ok(MsigAll {
-            multisig_scripts,
+        let native_scripts = (|| -> Result<_, DeserializeError> {
+            Ok(NativeScripts::deserialize(raw)?)
+        })().map_err(|e| e.annotate("native_scripts"))?;
+        Ok(ScriptAll {
+            native_scripts,
         })
     }
 }
 
-impl cbor_event::se::Serialize for MsigAny {
+impl cbor_event::se::Serialize for ScriptAny {
     fn serialize<'se, W: Write>(&self, serializer: &'se mut Serializer<W>) -> cbor_event::Result<&'se mut Serializer<W>> {
         serializer.write_array(cbor_event::Len::Len(2))?;
         self.serialize_as_embedded_group(serializer)
     }
 }
 
-impl SerializeEmbeddedGroup for MsigAny {
+impl SerializeEmbeddedGroup for ScriptAny {
     fn serialize_as_embedded_group<'se, W: Write>(&self, serializer: &'se mut Serializer<W>) -> cbor_event::Result<&'se mut Serializer<W>> {
         serializer.write_unsigned_integer(2u64)?;
-        self.multisig_scripts.serialize(serializer)?;
+        self.native_scripts.serialize(serializer)?;
         Ok(serializer)
     }
 }
 
-impl Deserialize for MsigAny {
+impl Deserialize for ScriptAny {
     fn deserialize<R: BufRead + Seek>(raw: &mut Deserializer<R>) -> Result<Self, DeserializeError> {
         (|| -> Result<_, DeserializeError> {
             let len = raw.array()?;
-            let ret = Self::deserialize_as_embedded_group(raw, len);
+            let mut read_len = CBORReadLen::new(len);
+            read_len.read_elems(2)?;
+            let ret = Self::deserialize_as_embedded_group(raw, /*mut read_len, */len);
             match len {
-                cbor_event::Len::Len(_) => /* TODO: check finite len somewhere */(),
+                cbor_event::Len::Len(_) => read_len.finish()?,
                 cbor_event::Len::Indefinite => match raw.special()? {
-                    CBORSpecial::Break => /* it's ok */(),
+                    CBORSpecial::Break => (),
                     _ => return Err(DeserializeFailure::EndingBreakMissing.into()),
                 },
             }
             ret
-        })().map_err(|e| e.annotate("MsigAny"))
+        })().map_err(|e| e.annotate("ScriptAny"))
     }
 }
 
-impl DeserializeEmbeddedGroup for MsigAny {
-    fn deserialize_as_embedded_group<R: BufRead + Seek>(raw: &mut Deserializer<R>, len: cbor_event::Len) -> Result<Self, DeserializeError> {
+impl DeserializeEmbeddedGroup for ScriptAny {
+    fn deserialize_as_embedded_group<R: BufRead + Seek>(raw: &mut Deserializer<R>, /*/*read_len: &mut CBORReadLen, */*/len: cbor_event::Len) -> Result<Self, DeserializeError> {
```

Comment:
The `DeserializeEmbeddedGroup` trait from before and now (after deserialize length checks on array/maps) are incompatible so I just commented out those parts here (and in other parts in this file) that interfered with it. If we re-generate the other parts of the library with those checks/other changes we could uncomment them out.

The newer code should generate less trait impls (only does DeserializeEmbeddedGroup when necessary - not often, instead of always) as well but I'm not sure how that translates to wasm size during packing, since there's also going to be the additional length checks.